src/Http/Http.Abstractions/src/Microsoft.AspNetCore.Http.Abstractions.csproj

```diff
@@ -30,6 +30,26 @@ Microsoft.AspNetCore.Http.HttpResponse</Description>
 
   <ItemGroup>
     <InternalsVisibleTo Include="Microsoft.AspNetCore.Http.Abstractions.Microbenchmarks" />
+    <InternalsVisibleTo Include="Microsoft.AspNetCore.Http.Extensions" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="RouteHandlerInvocationContextOfT.tt">
+      <Generator>TextTemplatingFileGenerator</Generator>
+      <LastGenOutput>RouteHandlerInvocationContextOfT.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
```

Comment:
I wonder if we can include this in the global directory.props

cc @dougbu

Reply:
You mean notice a project contains a .tt file and add the `<Service />` element @davidfowl❔ That may work but VS would rewrite it the next time someone opens the project IIRC.

We only have 3 projects containing this `<Service />` after this PR. How big an issue is this❔

Reply:
It's not an issue, I would just like to clean it up as we plan to code generate more things.

Reply:
It's up to you @davidfowl whether to centralize things in this PR. I recommend testing if the code in our root Directory.Build.props prevents VS rewriting the project to list the `<Service/>`. If "Save All" is fine after adding a new .tt file, you're good to go. Otherwise, the extra code will become redundant and something we'll end up fighting.

The specific approach would be something like

``` msbuild
<ItemGroup>
  <_GeneratorSources Include="$(MSBuildProjectDirectory)\**\*.tt" />
</ItemGroup>
<ItemGroup>
  <Service Condition=" '@(_GeneratorSources->Count()) ' != '0' " Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
</ItemGroup>
```